lib/collection/src/collection/point_ops.rs

```diff
@@ -340,42 +340,26 @@ impl Collection {
-                        records.into_iter().map(|mut record| {
-                            let value;
-                            if local_only {
-                                value = record.order_value.unwrap_or_else(|| {
-                                    order_by.get_order_value_from_payload(record.payload.as_ref())
-                                });
-                            } else {
-                                value = if let Some(order_value) = record.order_value {
-                                    order_by
-                                        .remove_order_value_from_payload(record.payload.as_mut());
-                                    order_value
-                                } else {
-                                    order_by
-                                        .remove_order_value_from_payload(record.payload.as_mut())
-                                };
-                                if !with_payload_interface.is_required() {
-                                    // Use None instead of empty hashmap
-                                    record.payload = None;
-                                }
-                            };
-                            (value, record)
-                        })
-                    })
                     // Get top results
-                    .kmerge_by(|(value_a, record_a), (value_b, record_b)| {
-                        match order_by.direction() {
-                            Direction::Asc => (value_a, record_a.id) < (value_b, record_b.id),
-                            Direction::Desc => (value_a, record_a.id) > (value_b, record_b.id),
-                        }
+                    .kmerge_by(|a, b| match order_by.direction() {
+                        Direction::Asc => (a.order_value, a.id) < (b.order_value, b.id),
+                        Direction::Desc => (a.order_value, a.id) > (b.order_value, b.id),
                     })
                     // Only keep the point with the most "valuable" order value
-                    .dedup_by(|(_, record_a), (_, record_b)| record_a.id == record_b.id)
-                    .map(|(_, record)| api::rest::Record::from(record))
+                    // This means that the same point will be repeated only if there is an alternating order value with another point
+                    // E.g.
+                    // For these points
+                    // { id: 1, values: [11, 21]}
+                    // { id: 2, values: [12, 22, 32]}
+                    //
+                    // The output will be
+                    // { id: 1, order_value: 11 }
+                    // { id: 2, order_value: 12 }
+                    // { id: 1, order_value: 21 }
+                    // { id: 2, order_value: 22 }
+                    // (no order_value 32)
+                    .dedup_by(|record_a, record_b| record_a.id == record_b.id)
```

Comment:
Is this desirable behavior?

Could we other wise dedup by the tuple `(record_a.id, record_a.order_value)`?

Reply:
I have the same opinion, here I only documented the current behavior.

IIRC, the intention for this is that the case when a point has many values, to only use one of them if they are consecutive, and there is no interleaving with another point. But tbh, it looks more like a bug, like it puzzled even ourselves.

@generall thoughts?

Reply:
Thanks. I think this is better!